pkg/controllers/management/kontainerdrivermetadata/metadata_handler.go

```diff
@@ -74,52 +98,56 @@ func Register(ctx context.Context, management *config.ManagementContext) {
 }
 
 func (m *MetadataController) sync(key string, setting *v3.Setting) (runtime.Object, error) {
-	if setting == nil || (setting.Name != rkeMetadataURL && setting.Name != rkeMetadataInterval) {
+	if setting == nil || (setting.Name != rkeMetadataConfig) {
 		return nil, nil
 	}
 
-	var interval time.Duration
-	if tickerData == nil {
-		if err := m.initAndStartTicker(); err != nil {
-			return nil, err
-		}
-		logrus.Infof("driverMetadata initialized successfully")
-		return setting, nil
-	}
-
-	if setting.Name == rkeMetadataURL {
-		currURL := settings.RkeMetadataURL.Get()
-		if currURL != prevURL {
-			if err := m.refresh(); err != nil {
-				return nil, err
-			}
-			prevURL = currURL
-		}
-		return setting, nil
+	settingValues, err := getSettingValues()
+	if err != nil {
+		return nil, err
 	}
 
-	interval, err := getTimeSetting()
+	interval, err := parseTime(settingValues[refreshInterval])
```

Comment:
I think it makes sense to do parseTime check in a store create/update in addition to this as well. So even if the time is negative/zero controller won't keep returning error

Reply:
we don't keep returning error if time is negative/zero. but adding validation in store seems like a good idea, would add it in a separate pr?